### PR TITLE
Fix/sentry issues 10 24

### PR DESCRIPTION
**Pull request by poocart**
Fixes couple issues:
- Do not crash app if by any mean account comes as undefined object (Sentry issue).
- Reload Assets screen even if not focused, but active account prop has changed – this was an issue on faster devices that caused Assets screen to stay as Legacy wallet after swtiching from it to Smart Wallet or PPN screens.

### CODE REVIEW

File: src/screens/Assets/Assets.js [hunk 1]
```diff
@@ -152,8 +152,9 @@ class AssetsScreen extends React.Component<Props, State> {
   }
 
   shouldComponentUpdate(nextProps: Props, nextState: State) {
+    const { activeAccount } = this.props;
     const isFocused = this.props.navigation.isFocused();
-    if (!isFocused) {
+    if (!isFocused && isEqual(nextProps.activeAccount, activeAccount)) {
```
Comment (jsidorenko) [inline src/screens/Assets/Assets.js]: It's faster to compare account ids rather than 2 objects here

File: src/screens/Accounts/Accounts.js [hunk 2]
```diff
@@ -475,6 +499,9 @@ class AccountsScreen extends React.Component<Props, State> {
 
     const walletsInList = isLegacyUser ? walletsToShow : [smartAccountCard];
 
+    // in case if undefined object comes to array (ref: Sentry issue ID 1292819194)
+    const accountsList = [...walletsInList, ...networksToShow].filter(item => !isEmpty(item));
```
Comment (jsidorenko) [inline src/screens/Accounts/Accounts.js]: How could you get `undefined` here?

File: src/screens/Accounts/Accounts.js [hunk 3]
```diff
@@ -39,24 +43,46 @@ import { ScrollWrapper } from 'components/Layout';
...
+  Rates,
+} from 'models/Asset';
+import type {
+  Account,
+  Accounts,
+} from 'models/Account';
+import type {
+  Dispatch,
+  RootReducerState,
+} from 'reducers/rootReducer';
```
Comment (jsidorenko) [inline src/screens/Accounts/Accounts.js]: I think it became less readable now

File: src/screens/Accounts/Accounts.js (hunk 2 above)
Comment (poocart) [reply src/screens/Accounts/Accounts.js]: As per PR description - we're not sure yet. If we put undefined here manually we get exactly same crash as in Sentry, the issue itself is yet to debug.

File: src/screens/Accounts/Accounts.js (hunk 3 above)
Comment (poocart) [reply src/screens/Accounts/Accounts.js]: Well, TBH it was auto lint fixes, but will revert to one line imports.

File: src/screens/Accounts/Accounts.js (hunk 2 above)
Comment (poocart) [reply src/screens/Accounts/Accounts.js]: @vyvil finally found an issue, will make another commit re this

**Pull request merged**